plugins/ui-update/utils/index.ts

```diff
@@ -0,0 +1,228 @@
+import { BehaviorSubject, Observable, of } from "rxjs";
+import { catchError, delay, take } from "rxjs/operators";
+import { graphqlObservable } from "@dcos/data-service";
+import gql from "graphql-tag";
+import semver, { SemVer } from "semver";
+
+import { schema as cosmosSchema } from "#SRC/js/data/cosmos";
+import { Package } from "#SRC/js/data/cosmos/Package";
+import { default as uiServiceSchema } from "#SRC/js/data/ui-update";
+import { UIMetadata } from "#SRC/js/data/ui-update/UIMetadata";
+import { UIAction, UIActions } from "#PLUGINS/ui-update/types/UIAction";
+import { FormattedPackageVersion } from "#PLUGINS/ui-update/types/FormattedPackageVersion";
+
+interface MutationError {
+  message: string;
+  name: string;
+}
+interface UIUpdateResult {
+  data: { updateDCOSUI: string };
+}
+interface UIRollbackResult {
+  data: { resetDCOSUI: string | null };
+}
+const FALLBACK_UI_METADATA: UIMetadata = {
```

Comment:
It strikes me that this may not be the best name or location for this `const` now. It made sense when I added for the error case in `queryUIServiceForMetadata`, but I ended up re-using it in a few other places after that.

I'll think about this and likely move it, open to suggestions here.